server/src/main/java/org/elasticsearch/index/mapper/XContentDataHelper.java

```diff
@@ -158,6 +175,10 @@ static void writeMerged(XContentBuilder b, String fieldName, List<BytesRef> enco
         b.endArray();
     }
 
+    public static boolean isDataPresent(BytesRef encoded) {
```

Comment:
Change this logic to check for empty and rename to `isEmpty`?

Reply:
Why is that better?

Reply:
All collections work with the concept of "being empty" and have an `isEmpty` and I think also that usually writing logic like "if it is empty then exit or return" is easier to understand...no data, nothing to do.

Reply:
Yes, `filter` is the reason.